beacon-chain/rpc/service.go

```diff
@@ -117,12 +131,35 @@ func (s *Service) SignBlock(ctx context.Context, req *pb.SignRequest) (*pb.SignR
 
 // LatestBeaconBlock streams the latest beacon chain data.
 func (s *Service) LatestBeaconBlock(req *empty.Empty, stream pb.BeaconService_LatestBeaconBlockServer) error {
-	// TODO: implement.
-	return errors.New("unimplemented")
+	// Right now, this streams every announced block received via p2p. It should only stream
+	// finalized blocks that are canonical in the beacon node after applying the fork choice
+	// rule.
+	for {
+		select {
+		case block := <-s.announcer.CanonicalBlockAnnouncement():
+			log.Info("Sending block to RPC clients")
+			if err := stream.Send(block.Proto()); err != nil {
+				return err
+			}
+		case <-s.ctx.Done():
+			return nil
+		}
+	}
 }
 
 // LatestCrystallizedState streams the latest beacon crystallized state.
 func (s *Service) LatestCrystallizedState(req *empty.Empty, stream pb.BeaconService_LatestCrystallizedStateServer) error {
-	// TODO: implement.
-	return errors.New("unimplemented")
+	// Right now, this streams every newly created crystallized state but should only
+	// stream canonical states.
+	for {
+		select {
+		case state := <-s.announcer.CanonicalCrystallizedStateAnnouncement():
+			log.Info("Sending state to RPC clients")
```

Comment:
Can we say `Sending crystallized state to RPC clients` instead? state could be active or crystallized and it could be confusing